javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -1283,9 +1254,162 @@ public String stringPrefix() {
+                int depthShift = this.depthShift, offset = this.offset;
+                if (isFullLeft()) {
+                    array = copyUpdate(emptyArray(), branchingFactor() - 1, array);
+                    depthShift += BRANCHING_BASE;
+                    offset = treeSize(branchingFactor() - 1, depthShift);
+                }
+
+                offset -= 1;
+                array = modifyLeafValue(array, depthShift, offset, leading);
+                return new VectorTree<>(array, offset, newSize, depthShift);
+            }
+        }
+
+        private boolean isFullLeft() { return offset == 0; }
+
+        VectorTree<T> appendLeaf(T trailing) {
+            final int newSize = length() + 1;
+            if (length() == 0) {
+                return new VectorTree<>(asArray(trailing), offset, newSize, depthShift);
+            } else {
+                Object[] array = this.array;
+                int depthShift = this.depthShift;
+                if (isFullRight(newSize)) {
+                    array = asArray(array);
+                    depthShift += BRANCHING_BASE;
+                }
+
+                array = modifyLeafValue(array, depthShift, offset + length(), trailing);
+                return new VectorTree<>(array, offset, newSize, depthShift);
+            }
+        }
+
+        private boolean isFullRight(int newSize) { return (offset + newSize) > treeSize(branchingFactor(), depthShift); }
+
+        VectorTree<T> update(int index, T element) {
+            final Object[] root = modifyLeafValue(array, depthShift, offset + index, element);
+            return new VectorTree<>(root, offset, length(), depthShift);
+        }
+
+        VectorTree<T> drop(int n) {
+            if (n <= 0) {
+                return this;
+            } else if (n >= length()) {
+                return emptyTree();
+            } else {
+                final Object[] root = modifyLeaf(array, depthShift, offset + n, Arrays::copyDrop, Arrays::copyDrop);
+                return new VectorTree<>(root, offset + n, length() - n, depthShift);
+            }
+        }
+
+        VectorTree<T> take(int n) {
+            if (n >= length()) {
+                return this;
+            } else if (n <= 0) {
+                return emptyTree();
+            } else {
+                final Object[] root = modifyLeaf(array, depthShift, (offset + n) - 1, Arrays::copyTake, Arrays::copyTake);
+                return new VectorTree<>(root, offset, n, depthShift);
+            }
+        }
+
+        private static Object[] modifyLeafValue(Object[] array, int depthShift, int index, Object element) {
+            final NodeModifier node = (o, i) -> copy(o, i + 1);
+            final NodeModifier leaf = (o, i) -> copyUpdate(o, i, element);
+            return modifyLeaf(array, depthShift, index, node, leaf);
+        }
+
+        private static Object[] modifyLeaf(Object[] array, int depthShift, int index, NodeModifier nodeModifier, NodeModifier leafModifier) {
```

Comment:
There's a measurable difference this way :dancer: 
Thanks @danieldietrich!

before:
```java
Update    slang_persistent/scala_persistent  1.03×  0.93×  1.17×
Prepend   slang_persistent/scala_persistent  0.64×  0.58×  0.44×
Append    slang_persistent/scala_persistent  0.84×  0.33×  0.26×
Slice     slang_persistent/scala_persistent  0.74×  1.19×  1.47×
```
after:
```java
Operation  Ratio                                 32   1024  32768 
Update     slang_persistent/scala_persistent  1.04×  1.14×  1.24×
Prepend    slang_persistent/scala_persistent  0.64×  0.58×  0.43×
Append     slang_persistent/scala_persistent  0.94×  0.37×  0.29×
Slice      slang_persistent/scala_persistent  0.79×  1.23×  1.64×
```